mbed-hal/spi_api.h

```diff
@@ -64,43 +97,64 @@ extern "C" {
 /** Initialize the SPI peripheral
  *
  * Configures the pins used by SPI, sets a default format and frequency, and enables the peripheral
+ * @note The function asserts that the selected pins map onto one SPI peripheral!
+ *
  * @param[out] obj  The SPI object to initialize
  * @param[in]  mosi The pin to use for MOSI
  * @param[in]  miso The pin to use for MISO
  * @param[in]  sclk The pin to use for SCLK
+ *
+ * @retval  `SPI_RC_OK`
  */
-void spi_init(spi_t *obj, PinName mosi, PinName miso, PinName sclk);
+spi_return_code_t spi_init(spi_t *obj, PinName mosi, PinName miso, PinName sclk);
```

Comment:
This should also take the CS pin optionally used when hardware supports driving it